docs/DevEnvironment.md

```diff
@@ -0,0 +1,158 @@
+<!--
+# @title Development Environment
+-->
+# Development Environment
+
+## Introduction
+
+Rouge is written in Ruby and has a number of development dependencies. To
+develop new features for Rouge (like a lexer for a new syntax) or to fix bugs,
+you need to set up a development environment.
+
+### Ruby and Git
+
+First thing's first. This guide is _not_ a guide to installing Ruby and Git.
+There are a number of excellent resources out there explaining how to do these
+things. For Ruby, we recommend [the official documentation][rb-inst-docs] and
+for Git, [GitHub's documentation][gh-inst-docs].
+
+[rb-inst-docs]: https://www.ruby-lang.org/en/documentation/installation/
+[gh-inst-docs]: https://help.github.com/en/articles/set-up-git
+
+### Terminal Emulators
+
+This guide assumes you are familiar with the _command line_. The command line is
+accessed through a _terminal emulator_. In macOS, the default emulator is called
+"Terminal" and can be found by searching for "Terminal" in Spotlight. In
+Windows, depending on your version, you may want to install a third-party
+emulator like [Cygwin][]. In Linux, well, you're probably reading this at the
```

Comment:
```suggestion
emulator like [Cygwin][] or just use the native `cmd.exe`. In Linux, well
you're probably reading this at the
```

Reply:
I'm not familiar enough with Windows to know whether trying to use Ruby and Git through `cmd.exe` is just asking for a world of pain. I gather it's not that bad?

Reply:
Couldn't say, even though i'm primarily a windows person, I just kinda automatically installed a linux vm to do some dev on it today assuming it was the right way to go ( and since i wanted to test my changes by copying the gem into a gitlab omnibus install )

Reply:
I use Ruby and Git on `cmd.exe` from Windows 7. The only thing I needed was the corresponding installers from [Rubyinstaller.org](https://rubyinstaller.org/) and [Git SCM](https://git-scm.com/download/win) and follow the installation instructions as documented in the respective sources. But all that is beyond the scope of this documentation here.

If at all an emulator, the better choice would be to opt for Linux subsystem on Windows 10 (WSL and upcoming WSL2).

Reply:
Good to know—will update to have a reference to `cmd.exe`.